assets/js/googlesitekit/components-gm3/Checkbox.js

```diff
@@ -16,6 +16,151 @@
  * limitations under the License.
  */
 
-export default function Checkbox() {
-	return null;
+/**
+ * External dependencies
+ */
+import '@material/web/checkbox/checkbox';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import Spinner from '../../components/Spinner';
+
+/**
+ * Internal dependencies
+ */
+import { getLabelFromChildren } from '../../util/react-children-to-text';
+
+export default function Checkbox( {
+	onChange,
+	id,
+	name,
+	value,
+	checked,
+	disabled,
+	children,
+	tabIndex,
+	onKeyDown,
+	loading,
+} ) {
+	const ref = useRef( null );
+
+	useEffect( () => {
+		const { current } = ref;
+
+		if ( ! current ) {
+			return;
+		}
+
+		function updateCheckedState() {
+			current.checked = checked;
+
+			// The checked property doesn't get reflected to the inner input element checked state,
+			// so we need to set it manually.
+			const innerInput = current.shadowRoot.querySelector( 'input' );
+			if ( innerInput ) {
+				innerInput.checked = checked;
+			}
+		}
+
+		function handleChange( event ) {
+			onChange?.( event );
+
+			// Restore current checked state for controlled component behaviour.
+			updateCheckedState();
+		}
+
+		current.addEventListener( 'change', handleChange );
+
+		if ( onKeyDown ) {
+			current.addEventListener( 'keydown', onKeyDown );
+		}
+
+		updateCheckedState();
+
+		return () => {
+			current.removeEventListener( 'change', handleChange );
+
+			if ( onKeyDown ) {
+				current.removeEventListener( 'keydown', onKeyDown );
+			}
+		};
+	}, [ checked, onChange, onKeyDown ] );
+
+	// TODO: Remove `onLabelClick` once the `md-checkbox` component is updated to be a form-associated custom element.
+	const onLabelClick = () => {
+		const { current } = ref;
+
+		if ( ! current || disabled ) {
+			return;
+		}
+
+		current.checked = ! current.checked;
+
+		current.dispatchEvent(
+			new Event( 'change', { bubbles: true, cancelable: true } )
+		);
+	};
+
+	return (
+		<div className="googlesitekit-component-gm3__checkbox">
+			{ loading && (
+				<div className="googlesitekit-component-gm3__checkbox--loading">
+					<Spinner isSaving />
+				</div>
+			) }
+			{ ! loading && (
+				<md-checkbox
+					id={ id }
+					ref={ ref }
+					role="checkbox"
+					aria-label={ getLabelFromChildren( children ) }
```

Comment:
As mentioned, I've added this `getLabelFromChildren` utility to accommodate our existing use cases where the label is defined as a React element rather than a string. This allows the component to retain the same props shape and it can be swapped directly in to replace the existing GM2 Checkbox.

Alternatively we could add a new prop instead, e.g. `ariaLabel`, which we'd then need to make sure to populate when it comes to swapping to the new Checkbox... What do you think?

Reply:
I think we _should_ support `aria-label` as a prop in general (part of why I wondered about the limited props available to this component), but with the note that this issue is about maintaining/starting with GM2 compatibility, I think this is a great start.

I would expect the behaviour you've implemented, and then having an optional `aria-label` as an override if needed. Since we won't need that for now, I think this is the best option for this issue 👍🏻